### PR TITLE
fix: add autoSize to Combobox positioning

### PR DESCRIPTION
**Pull request by smhigley**
## Previous Behavior

When the Combobox popup is outside the bounds of the viewport, keyboard users can arrow to an option that isn't visible. Since arrowing also moves between options rather than scrolling, there isn't a good way to get the option into view.

I'm adding this specifically to Combobox (and the other related components that use the `react-combobox` hooks) because unlike Menu and other popups, it doesn't actually move keyboard focus into the popup. Arrowing to an out-of-view MenuItem will cause the page to scroll, since keyboard focus is moving out of view.

We might consider adding this by default to Menu as well, but it's more clearly needed in Combobox because of focus remaining on the trigger element.

## New Behavior

Combobox's popup has `autoSize: true`, which forces it to confine itself to the viewport bounds.

## Related Issue(s)

- Fixes [ADO issue](https://dev.azure.com/microsoftdesign/fluent-ui/_workitems/edit/18642)

### CODE REVIEW

**Review by sopranopillow — APPROVED** [review]

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31864/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">27</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31864/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">37</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31864/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31864/perf-test-react-components/Avatar-mount_base.html">618</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31864/perf-test-react-components/Avatar-mount.html">619</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31864/perf-test-react-components/Button-mount_base.html">291</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31864/perf-test-react-components/Button-mount.html">304</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31864/perf-test-react-components/Field-mount_base.html">1121</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31864/perf-test-react-components/Field-mount.html">1103</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31864/perf-test-react-components/FluentProvider-mount_base.html">694</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31864/perf-test-react-components/FluentProvider-mount.html">708</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31864/perf-test-react-components/FluentProviderWithTheme-mount_base.html">76</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31864/perf-test-react-components/FluentProviderWithTheme-mount.html">78</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31864/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">27</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31864/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">37</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31864/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31864/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">77</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31864/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">79</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31864/perf-test-react-components/MakeStyles-mount_base.html">855</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31864/perf-test-react-components/MakeStyles-mount.html">874</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31864/perf-test-react-components/Persona-mount_base.html">1804</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31864/perf-test-react-components/Persona-mount.html">1740</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31864/perf-test-react-components/SpinButton-mount_base.html">1443</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31864/perf-test-react-components/SpinButton-mount.html">1418</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31864/perf-test-react-components/SwatchPicker-mount_base.html">1663</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31864/perf-test-react-components/SwatchPicker-mount.html">1707</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports | Baseline (minified/GZIP) | PR    | Change     |
| :---------------- | -----------------------: | ----: | ---------: |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox (including child components)</abbr> | `103.559 kB`<br />`33.975 kB` | `103.571 kB`<br />`33.978 kB` | `12 B` <br />`3 B` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Dropdown.fixture.js'>Dropdown (including child components)</abbr> | `104.145 kB`<br />`33.884 kB` | `104.157 kB`<br />`33.888 kB` | `12 B` <br />`4 B` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ReactComponents.fixture.js'>react-components: entire library</abbr> | `1.094 MB`<br />`270.787 kB` | `1.094 MB`<br />`270.794 kB` | `12 B` <br />`7 B` |
| <samp>react-timepicker-compat</samp> <br /> <abbr title='bundle-size/TimePicker.fixture.js'>TimePicker</abbr> | `106.547 kB`<br />`35.513 kB` | `106.559 kB`<br />`35.517 kB` | `12 B` <br />`4 B` |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports | Size (minified/GZIP) |
| ----------------- | -------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> | `69.141 kB`<br />`20.157 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `211.741 kB`<br />`60.957 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `44.442 kB`<br />`14.607 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr> | `8.39 kB`<br />`2.64 kB` |
| <samp>react-tag-picker</samp> <br /> <abbr title='bundle-size/Package.fixture.js'>@fluentui/react-tag-picker - package</abbr> | `181.187 kB`<br />`54.484 kB` |
</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/1d0e64be70442fae5b31c57ff69053580d63bfeb'>1d0e64be70442fae5b31c57ff69053580d63bfeb</a></sub>
<div id="bundleSizeComment9423"/>

**Review by khmakoto — APPROVED** [review]

File: change/@fluentui-react-combobox-49a36618-f887-4534-a022-f6b4fd32d199.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-combobox-49a36618-f887-4534-a022-f6b4fd32d199.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=31864&commitId=ab9ee64655718d52f738bddbf001c98344b3a870&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.badgeMask - RTL.normal.chromium.png|0| Added|
|Avatar Converged.size+inactive+badge.normal.chromium.png|0| Removed|

</details>


  <details><summary><b>SwatchPicker Converged</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|SwatchPicker Converged.default - RTL.default.chromium.png|0| Added|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

Comment (smhigley) [reply change/@fluentui-react-combobox-49a36618-f887-4534-a022-f6b4fd32d199.json:1]: not related to this PR

**Pull request merged**